### PR TITLE
Fix multiplatform support

### PR DESCRIPTION
**Pull request by tonilopezmr**
### :tophat: What is the goal?

* Fix Kotlin Multiplatform environments, it doesn't generate `HaguConfig` in the common module.

### How is it being implemented?

* Add kotlin plugin to access to Kotlin compilation directories depending on the Kotlin environment, simple Kotlin or Multiplatform
* Add generated sources into classpath compilation, **it still needs a Gradle configuration because IntelliJ doesn't recognize the generated code** 
* Multiplatform example with test to make sure the plugin is working with Kotlin Multiplatform

> I'd like to remove this Gradle configuration: 

>```
>kotlin.sourceSets["commonMain"].kotlin.srcDirs("$buildDir/generated/kotlin/config")
>```

> At this moment I don't know how IntelliJ can recognize it without that Gradle line.

* Changed plugin artifact name to `gradle-plugin` and added `com.karumi` for apply the plugin.

### How can it be tested?

🤖

### CODE REVIEW

File: plugin/src/main/kotlin/com/karumi/hagu/plugin/HaguPlugin.kt [hunk 1]
```diff
@@ -1,20 +1,41 @@
...
+import com.karumi.hagu.plugin.source.sources
+import java.io.File
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType
 
 class HaguPlugin : Plugin<Project> {
 
+  companion object {
+    private const val HAGU_GENERATED_SOURCE_FOLDER = "generated/kotlin/config"
```
Comment (pedrovgs) [inline plugin/src/main/kotlin/com/karumi/hagu/plugin/HaguPlugin.kt:12]: Is this the way constants are declared in Kotlin? I thought we should use ``haguGeneratedSourceFolder``. I don't think this is important, but I'm just curious about the official codestyle.

Comment (pedrovgs) [reply plugin/src/main/kotlin/com/karumi/hagu/plugin/HaguPlugin.kt:12]: It seems to be the expected way! https://kotlinlang.org/docs/reference/coding-conventions.html#property-names

**Review by pedrovgs — APPROVED** [review]
@tonilopezmr please review [this PR](https://github.com/Karumi/Hagu/pull/9) before merging this one.

File: plugin/src/main/kotlin/com/karumi/hagu/plugin/source/SourceRoots.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        sourceDirectorySet = compilation.defaultSourceSet.kotlin,
+        sourceSets = compilation.allKotlinSourceSets.map { it.name },
+        registerTaskDependency = { task ->
+          (target as? KotlinNativeTarget)?.binaries?.forEach {
+            it.linkTask.dependsOn(task)
+          }
+          compilation.compileKotlinTask.dependsOn(task)
+        }
+      )
+    }
```
Comment (Serchinastico) [inline plugin/src/main/kotlin/com/karumi/hagu/plugin/source/SourceRoots.kt]: I'm seeing too much nesting with maps and flatmaps. We can maybe extract a function that gets the `Source` instance from a compilation and leave it like:

```kotlin
targets.flatMap { target ->
    target.compilations.mapNotNull { it.toSource() } 
}
```

**Review by Serchinastico — APPROVED** [review]
This is looking great, my only concern is about style so it's not a blocker :+1:

File: plugin/src/main/kotlin/com/karumi/hagu/plugin/source/SourceRoots.kt (hunk 2 above)
Comment (tonilopezmr) [reply plugin/src/main/kotlin/com/karumi/hagu/plugin/source/SourceRoots.kt]: Yes, I can improve it!

**Pull request merged**